spec/io/sql_data_source_spec.rb

```diff
@@ -14,70 +14,58 @@
     ActiveRecord::Base.connection
   end
 
+  let(:dat_file) do
+    'spec/fixtures/bank2.dat'
+  end
+
   let(:query) do
     'select * from accounts'
   end
 
+  let(:source) do
+    active_record_connection
+  end
+
   describe '.make_dataframe' do
-    context 'with DBI::DatabaseHandle' do
-      it 'returns a dataframe' do
-        result = Daru::IO::SqlDataSource.make_dataframe(dbi_handle, query)
-        expect(result).to be_a(Daru::DataFrame)
-        expect(result.nrows).to eq(2)
-        expect(result.row[0][:id]).to eq(1)
-        expect(result.row[0][:name]).to eq('Homer')
-      end
+    subject(:df) { Daru::IO::SqlDataSource.make_dataframe(source, query) }
 
-      context 'with an object not a string as a query' do
-        it 'raises ArgumentError' do
-          expect {
-            Daru::IO::SqlDataSource.make_dataframe(dbi_handle, Object.new)
-          }.to raise_error(ArgumentError)
-        end
-      end
+    context 'with DBI::DatabaseHandle' do
+      let(:source) { dbi_handle }
+      it { is_expected.to be_a(Daru::DataFrame) }
+      it { expect(df.row[0]).to have_attributes(id: 1) }
+      it { expect(df.row[0]).to have_attributes(age: 20) }
```

Comment:
Join those two: `it { expect(df.row[0]).to have_attributes(id: 1, age: 20) }`, it is logically just one test "we have right row here".